Split/Matchers/Semver/Semver.swift

```diff
@@ -155,6 +103,82 @@ class Semver: Equatable {
         return lhs.version == rhs.version
     }
 
+    private func getAndRemoveMetadataIfExists(_ version: String) throws -> (metadata: String?, versionWithoutMetadata: String) {
+        var vWithoutMetadata = ""
+        var tMetadata = ""
+        if let index = version.firstIndex(of: kMetadataDelimiter) {
+            let metadataIndex = version.index(after: index)
+
+            if metadataIndex >= version.endIndex {
+                throw SemverParseError.invalidVersionFormat("Unable to convert to Semver, incorrect metadata")
+            }
+
+            tMetadata = try getMetadata(version, metadataIndex)
+
+            let versionSubstring = version[..<index]
+
+            vWithoutMetadata = String(versionSubstring)
+        } else {
+            vWithoutMetadata = version
+        }
+
+        return (tMetadata, vWithoutMetadata)
+    }
+
+    private func getMetadata(_ version: String, _ index: String.Index) throws -> String {
+        let metadataString = String(version[index...])
+
+        if metadataString.isEmpty {
+            throw SemverParseError.invalidVersionFormat("Unable to convert to Semver, incorrect metadata")
+        }
+
+        return metadataString
+    }
+
+    private func getAndRemovePreReleaseIfExists(_ vWithoutMetadata: String) throws -> (preRelease: [String], isStable: Bool, versionWithoutPreRelease: String) {
```

Comment:
Maybe you can use the model here too:
`func versionWithoutPreRelease(withoutMetaData: String) throws -> SemverModel {`